app/main.py

```diff
@@ -98,6 +130,22 @@ def create_presigned_url(expiration=3600):
     return response
 
 
+# Reverse proxy for objects from S3, a simple get object
+# operation. This is used by scaffoldvuer and its 
+# important to keep the relative <path> for accessing
+# other required files.
+@app.route("/s3-resource/<path:path>")
+def direct_download_url(path):
+    bucket_name = "blackfynn-discover-use1"
+    response = s3.get_object(
+        Bucket=bucket_name,
+        Key=path,
+        RequestPayer="requester",
+    )
+    resource = response["Body"].read()
+    return resource
```

Comment:
I'm curious about this endpoint: how large of files do you expect to proxy through it?

It looks like the entire object is read into memory here. I'm concerned that requests for larger files will crash the server when it runs out of memory.  (@JeremyLecoeur do you know how much memory is allocated to the Heroku server?)

Depending on the needs of the frontend there are a couple ways that this could be made safer:

1. If the frontend only needs to download small files (kilobytes up to a couple megabytes of data) we could put a strict limit on the size of files accessible via this endpoint. The `response` object has a `ContentLength` key that can be validated before the data is read.
2. For larger files (say a couple hundred megabytes, perhaps even a gigabyte) Flask could stream the response body without loading it all in memory (see https://flask.palletsprojects.com/en/1.1.x/patterns/streaming/). In this case it is probably still a good idea to cap the max file size. 
3. For anything larger it's probably best to serve the file directly from the AWS servers. This endpoint could generate and return a presigned URL that the frontend uses to load the object directly from S3 (https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-presigned-urls.html).

Reply:
I'm not 100% sure but I think a 5MB file would be rare for the requirements of this endpoint.

Reply:
@rlmv : 512MB